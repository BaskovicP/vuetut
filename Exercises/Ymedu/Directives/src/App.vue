<template>
  <div class="container">
    <div class="row">
      <div class="col-xs-12 col-sm-8 col-sm-offset-2 col-md-6 col-md-offset-3">
        <h1 v-myon:click="consoleLog('Don\'t you click on me!!!!')">
          Directives Exercise
        </h1>
        <h1 v-myon:mouseover="consoleLog('PLease do not touch me!!!')">
          Directives Exercise moody
        </h1>
        <!-- Exercise -->
        <!-- Build a Custom Directive which works like v-on (Listen for Events) -->
      </div>
    </div>
  </div>
</template>

<script>
export default {
  directives: {
    myon: {
      bind(el, binding, vnode) {
        el.addEventListener(binding.arg, binding.value);
      }
    }
  },
  methods: {
      // we use this method because vue prefixes all the function calls in template with _vm
    consoleLog(message) {
      return () => console.log(message);
    }
  }
};
</script>

<style></style>
