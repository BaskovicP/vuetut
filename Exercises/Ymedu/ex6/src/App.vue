<template>
  <div class="container">
    <Header></Header>
    <hr />
    <div class="row">
      <div class="col-xs-12 col-sm-6">
        <List></List>
      </div>
      <div class="col-xs-12 col-sm-6">
        <Details></Details>
      </div>
    </div>
    <hr />
    <div class="row">
      <div class="col-xs-12">
        <ServerManagement></ServerManagement>
      </div>
    </div>
  </div>
</template>

<script>
//Localy registered components later used in the main app components
import Details from "./components/Details.vue";
import Header from "./components/Header.vue";
import List from "./components/List.vue";
import ServerManagement from "./components/ServerManagement.vue";
export default {
  components: {
    Details,
    Header,
    List,
    ServerManagement
  }
};
</script>

<style></style>
