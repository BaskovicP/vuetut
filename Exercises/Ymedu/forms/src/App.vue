<template>
  <div class="container">
    <form>
      <div class="row">
        <div
          class="col-xs-12 col-sm-8 col-sm-offset-2 col-md-6 col-md-offset-3"
        >
          <!-- Exercise 1 -->
          <!-- Create a Signup Form where you retrieve the following Information -->
          <!-- Full Name (First Name + Last Name) -->
          <section v-if="!formSubmited">
          <div class="form-group">
            <label>First Name</label>
            <input
              id="first-name"
              type="text"
              class="form-control"
              v-model="firstName"
            />
          </div>
          <div class="form-group">
            <label>Last Name</label>
            <input
              id="last-name"
              type="text"
              class="form-control"
              v-model="lastName"
            />
          </div>
          <div class="form-group">
            <label>Mail</label>
            <input id="mail" type="email" class="form-control" v-model="mail" />
          </div>
          <div class="form-group">
            <label>Password</label>
            <input
              id="password"
              type="password"
              class="form-control"
              v-model="password"
            />
          </div>
          <div class="form-group">
            <label>Store data? </label>
            <label
              ><input type="radio" value="Yes" v-model="storeInDatabase" /> Yes
            </label>

            <label
              ><input type="radio" value="No" v-model="storeInDatabase" />No
            </label>
          </div>
          <button
          id="main-submit-button"
            class="primary"
            type="submit"
            @click.prevent="formSubmited = !formSubmited"
          >
            Sunmit all data here
          </button>
          </section>

          <!-- Mail -->
          <!-- Password -->
          <!-- Store Data? Yes/No -->

          <!-- Exercise 2 -->
          <!-- Only display the Form if it has NOT been submitted -->
          <!-- Display the Data Summary ONCE the Form HAS been submitted -->

          <!-- Exercise 3 -->
          <!-- Edit the Example from above and create a custom "Full Name" Control -->
          <!-- which still holds the First Name and Last Name Input Field -->
        </div>
      </div>
    </form>
    <hr />
    <div class="row" v-if="formSubmited">
      <div class="col-xs-12 col-sm-8 col-sm-offset-2 col-md-6 col-md-offset-3">
        <div class="panel panel-default">
          <div class="panel-heading">
            <h4>Your Data</h4>
          </div>
          <div class="panel-body">
            <p>Full Name: {{ fullName }}</p>
            <p>Mail:{{ mail }}</p>
            <p>Password:{{ password }}</p>
            <p>Store in Database?: {{ storeInDatabase }}</p>
          </div>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
export default {
  data() {
    return {
      firstName: "",
      lastName: "",
      mail: "",
      password: "",
      storeInDatabase: "Yes",
      formSubmited: false
    };
  },
  computed: {
    fullName() {
      return this.firstName + " " + this.lastName;
    }
  }
};
</script>

<style></style>
