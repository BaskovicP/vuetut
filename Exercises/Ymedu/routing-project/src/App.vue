<template>
  <div id="app">
    <router-link to="/user/123" tag="button" active-class="my-active" exact>
      UserPage</router-link
    >
    <br />
    <br />
    <button @click="leadMeToSomewhere">
      Imperative jump
    </button>
    <router-view></router-view>
  </div>
</template>

<script>
export default {
  name: "app",
  methods: {
    leadMeToSomewhere() {
      this.$router.push("/newPage");
    }
  }
};
</script>

<style>
.my-active {
  color: blue;
  border: 2px solid blue;
}
</style>
