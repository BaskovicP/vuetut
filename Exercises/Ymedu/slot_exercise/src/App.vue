<template>
  <div class="container">
    <div class="row">
      <div class="col-xs-12">
        <br />
        <button @click="color = 'appBlue'" class="btn btn-primary">
          Load Blue Template
        </button>
        <button @click="color = 'appGreen'" class="btn btn-success">
          Load Green Template
        </button>
        <button @click="color = 'appRed'" class="btn btn-danger">
          Load Red Template
        </button>
        <hr />
        <component :is="color">
          <p>
            Some random text that will be inserted into the default component
            slot
          </p>
          <h1 slot="heading">This beauty is {{ color }}</h1>
        </component>
      </div>
    </div>
  </div>
</template>

<script>
import Blue from "./components/Blue.vue";
import Green from "./components/Green.vue";
import Red from "./components/Red.vue";

export default {
  data: () => ({
    color: "appBlue"
  }),
  components: {
    appBlue: Blue,
    appGreen: Green,
    appRed: Red
  }
};
</script>

<style></style>
