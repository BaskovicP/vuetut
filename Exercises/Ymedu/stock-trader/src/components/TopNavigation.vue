<template>
  <div id="top-navigation-wrapper">
    <nav class="navbar navbar-expand-lg navbar-light bg-light">
      <div class="navbar-header">
        <router-link
          class="navbar-brand"
          to="/"
          tag="a">
          StockTrader
        </router-link>
      </div>
      <button
        class="navbar-toggler"
        type="button"
        ata-toggle="collapse"
        data-target="#navbarSupportedContent"
        aria-controls="navbarSupportedContent"
        aria-expanded="false" aria-label="Toggle navigation">
        <span class="navbar-toggler-icon"></span>
      </button>

      <div id="navbarSupportedContent" class="collapse navbar-collapse">
        <ul class="navbar-nav mr-auto">
          <li class="nav-item">
            <router-link
              active-class="nav-link active"
              class="nav-link"
              to="/portfolio"
              tag="a">
              Portfolio
            </router-link>
            <!-- <a class="nav-link" href="#">Portfolio </a> -->
          </li>
          <li class="nav-item">
            <router-link
              active-class="nav-link active"
              class="nav-link"
              to="/stocks"
              tag="a">
              Stocks
            </router-link>
          </li>
          <li class="nav-item">
            <router-link
              active-class="nav-link active"
              class="nav-link"
              to="/analysis"
              tag="a">
              Analysis
            </router-link>
          </li>
          <li class="nav-item">
            <a
              @click="endDay"
              id="end-day-btn"
              class="nav-link"
              href="#">End Day</a>
          </li>
          <li class="nav-item">
            <div class="nav-link">
              Day number:
              <span
                id="day-number"
                style="color: black">{{ day }}</span>
            </div>
          </li>
        </ul>
        <span class="navbar-text">
          FUNDS: {{ funds | currency }}
        </span>
      </div>
    </nav>
  </div>
</template>

<script>

import { mapActions, mapGetters } from 'vuex';
export default {
  computed: {
    ...mapGetters({
      funds: 'funds',
      day: 'day'
    })
  },
  methods: {
    ...mapActions({
      endDay: 'randomizeStocks'
    })
  }
};
</script>

<style scoped>
#top-navigation-wrapper {
  margin-bottom: 1rem;
}
</style>
