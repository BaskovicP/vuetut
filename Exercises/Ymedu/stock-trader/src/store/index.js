import * as actions from './actions';
import portfolio from './modules/portfolio';
import stocks from './modules/stocks';
import Vue from 'vue';
import Vuex from 'vuex';

Vue.use(Vuex);

export default new Vuex.Store({
  actions,
  modules: {
    stocks,
    portfolio
  }
});
