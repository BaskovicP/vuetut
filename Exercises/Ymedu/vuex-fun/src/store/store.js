import Vue from "vue";
import Vuex from "vuex";

Vue.use(Vuex);

export const store = new Vuex.Store({
  state: {
    counter: 0
  },
  getters: {
    // name : state => function
    doubleCounter: state => state.counter * 2
  },
  mutations: {
    increment: (state, payload) => {
      state.counter += payload.by;
    },
    decrement: state => {
      state.counter--;
    }
  },
  actions: {
    increment: ({ commit }) => {
      commit("increment");
    },
    decrement: ({ commit }) => {
      commit("decrement");
    },
    asyncIncrement: ({ commit }, payload) => {
      setTimeout(() => commit("increment", payload), payload.time);
    }
  }
});
