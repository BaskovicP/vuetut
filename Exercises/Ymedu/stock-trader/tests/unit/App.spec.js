import { localVue, router } from '../Factory';
import App from '@/App';
import { mount } from '@vue/test-utils';
import store from '@/store/index';
import TopNavigation from '@/components/TopNavigation.vue';

describe('MainApp.vue', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(App, {
      store,
      router,
      localVue: localVue()
    });
  });

  it('contains the navigation bar component', () => {
    expect(wrapper.contains(TopNavigation)).toBe(true);
  });
  it('should render the component with a starting page', () => {
    expect(wrapper.text().match('Trade or View your Portfolio').length).toBe(1);
  });
  it('should change the pages with router', async () => {
    router.push('/portfolio');
    await wrapper.vm.$nextTick();
    expect((wrapper.text().match('Please buy some stocks') || []).length).toBe(1);
    router.push('/analysis');
    await wrapper.vm.$nextTick();
    expect((wrapper.text().match('End the current day. Please select a stock from the stock tab') || []).length).toBe(1);
    router.push('/stocks');
    await wrapper.vm.$nextTick();
    expect(wrapper.text()).toMatchSnapshot();
  });
});
