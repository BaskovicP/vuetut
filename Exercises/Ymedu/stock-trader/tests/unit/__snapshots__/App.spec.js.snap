// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainApp.vue should change the pages with router 1`] = `
"StockTrader  Portfolio  Stocks Analysis End Day Day number: 1 
        FUNDS: 10,000 HRK
       
        BMW
        (Price: 110)  
          Buy
         
          Analyze
        
        Google
        (Price: 200)  
          Buy
         
          Analyze
        
        Apple
        (Price: 250)  
          Buy
         
          Analyze
        
        Twitter
        (Price: 8)  
          Buy
         
          Analyze
        
        OPTE
        (Price: 8)  
          Buy
         
          Analyze
        
        HT
        (Price: 8)  
          Buy
         
          Analyze
        
        PODR
        (Price: 800)  
          Buy
         
          Analyze
        
        ULJN
        (Price: 2)  
          Buy
         
          Analyze
        
        ZABA
        (Price: 78)  
          Buy
         
          Analyze"
`;
